### PR TITLE
Setting back the old response of `/me/membership`

### PR DESCRIPTION
**Pull request by svillafe**
# Summary
In order to have backwards compatibility, this PR removes all the contributor information from the `/me/membership` endpoint and keeps it only in `/me`.
Additionally, the field `ContributionFrequency` has been renamed to `ContributionPaymentPlan` and the `contributor`  field inside content access object was change to recurringContributor.

This PR is a follow-up to https://github.com/guardian/members-data-api/pull/191

# Responses
Therefore the endpoints will return for a user who is a recurring contributor and a member:

### `/me/membership`

**Status Code:** `404`
```
{
    "message": "Not found",
    "details": "User was found but they are not a member",
    "statusCode": 404
}
```
### `/me`
**Status Code:** `200`
```
{
    "userId": "30000600",
    "contributionPaymentPlan": "Monthly Contribution",
    "contentAccess": {
        "member": false,
        "paidMember": false,
        "recurringContributor": true
    }
}
```
### `/me/membership`

```
{
    "userId": "30000602",
    "tier": "Supporter",
    "membershipNumber": "1234",
    "contentAccess": {
        "member": true,
        "paidMember": true
    }
}
```
### `/me`
```
{
    "userId": "30000602",
    "tier": "Supporter",
    "membershipNumber": "1234",
    "cardExpirationMonth": 12,
    "cardExpirationYear": 2042,
    "membershipJoinDate": "2017-06-20",
    "contentAccess": {
        "member": true,
        "paidMember": true,
        "recurringContributor": false
    }
}
```

### `/me/membership`
```
{
   "userId": "30000602",
   "tier": "Supporter",
   "membershipNumber": "324145",
   "contentAccess": {
      "member": true,
      "paidMember": true
   }
}
```
### `/me`
```
{
   "userId": "30000602",
   "tier": "Supporter",
   "membershipNumber": "324145",
   "cardExpirationMonth": 12,
   "cardExpirationYear": 2042,
   "contributionPaymentPlan": "Monthly Contribution",
   "membershipJoinDate": "2017-06-20",
   "contentAccess": {
      "member": true,
      "paidMember": true,
      "recurringContributor": true
   }
}
```

### CODE REVIEW

**Comment by svillafe** [comment]
After talking with @paulbrown1982 @davidfurey, we will rename the field `ContributionFrequency` to be `ContributionPaymentPlan`. Additionally, we will change the flag `Contributor` to become `RecurringContributor`

File: membership-attribute-service/app/models/Attributes.scala [hunk 1]
```diff
@@ -57,10 +60,55 @@ object Attributes {
...
+                       Tier: Option[String] = None,
+                       MembershipNumber: Option[String],
+                       AdFree: Option[Boolean] = None,
+                       ContentAccess : MembershipContentAccess)
+
+object MembershipAttributes {
+
+  implicit val jsWrite: OWrites[MembershipAttributes] = (
+    (__ \ "userId").write[String] and
+    (__ \ "tier").writeNullable[String] and
```
Comment (paulbrown1982) [inline membership-attribute-service/app/models/Attributes.scala]: tier should always be present on a Membership product.

Comment (svillafe) [inline membership-attribute-service/app/models/Attributes.scala]: Fair enough, in the case you hit this endpoint for an only contributor will return `tier:null`

Comment (paulbrown1982) [inline membership-attribute-service/app/models/Attributes.scala]: Also, please check therefore that the /me/membership endpoint does a 404 for a valid recurring contributor.

**Comment by paulbrown1982** [comment]
👍 LGTM. We will talk about name spacing or taking-the-soonest cardExpirationMonth/cardExpirationYear in another PR.

File: membership-attribute-service/app/models/Attributes.scala (hunk 1 above)
Comment (svillafe) [inline membership-attribute-service/app/models/Attributes.scala]: After testing, this endpoint will only return `200` if you are a member. See description above.

**Comment by paulbrown1982** [comment]
👍

**Comment by svillafe** [comment]
Teamcity failed to build the project due to a 500 in one of the dependencies. It is trying to build it again.

**Pull request merged**